### PR TITLE
Fixed- Allow only numbers in input field

### LINKED ISSUE #18: [Web- html,css,javascript] Allow only numbers in input field
**Title**
Allow only number keys in code input

**Project link**
https://github.com/DevilsAutumn/Hacktoberfest22/tree/main/Web/Html%2CCSS%2CJavascript/verify-account-ui

**Domain**
Add 'x' inside [ ] to cross check box.

- [ ] App with Flutter
- [ ] App with Java
- [ ] App with Kotlin
- [x] Web with Html,css javascript
- [ ] Web with React

**Please describe your feature.**
Code verification input boxes are allowing keys other than number also. It should not take input other than number key.

**Describe the solution you'd like**
NA

**Additional context**
May I solve this issue.

### PR DESCRIPTION
**Pull request by kunal00000**
# Description

Please include a summary of the change and which issue is fixed.

Fixes #18

Replace `issue_no` with the issue number which is fixed in this PR

## Type of change

_Please delete options that are not relevant._

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

File: Web/Html,CSS,Javascript/verify-account-ui/index.html [hunk 1]
```diff
@@ -11,12 +11,12 @@
       <h2>Verify Your Account</h2>
       <p>We emailed you the six digit code to cool_guy@email.com <br/> Enter the code below to confirm your email address.</p>
       <div class="code-container">
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
+        <input type="number" class="code" placeholder="0" min="0" max="9" maxlength="1" required>
```
Comment (DevilsAutumn) [inline Web/Html,CSS,Javascript/verify-account-ui/index.html:14]: maxlength is ignored for type = numbers as it counts the number of characters.

**Comment by kunal00000** [comment]
You can merge this branch now.

**Comment by DevilsAutumn** [comment]
I think it still needs improvement.
Screenshot 2022-10-05 173949

**Comment by kunal00000** [comment]
@DevilsAutumn review it once.

File: Web/Html,CSS,Javascript/verify-account-ui/index.html [hunk 2]
```diff
@@ -11,12 +11,12 @@
       <h2>Verify Your Account</h2>
       <p>We emailed you the six digit code to cool_guy@email.com <br/> Enter the code below to confirm your email address.</p>
       <div class="code-container">
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
-        <input type="number" class="code" placeholder="0" min="0" max="9" required>
+        <input type="text" class="code" placeholder="0" maxlength="1" required>
```
Comment (DevilsAutumn) [inline Web/Html,CSS,Javascript/verify-account-ui/index.html:14]: See this [comment](https://github.com/DevilsAutumn/Hacktoberfest22/pull/26#discussion_r986884434)

**Comment by DevilsAutumn** [comment]
> @DevilsAutumn review it once. on pressing keys faster it may leave boxes but while verifying codes we press it slowly. so have a look.

try this logic in your js file and see if it causes any error.
```
codes.forEach((code, idx) => {
  code.addEventListener("keydown", (e) => {
    if (e.keyCode >= 48 && e.keyCode <= 57 && codes[idx].value == "") {
      setTimeout(() => codes[idx + 1].focus(), 10);
    } else if (e.key === "Backspace") {
      codes[idx - 1].value = "";
      setTimeout(() => codes[idx - 1].focus(), 10);
    } else if (e.key === "Enter") {
      for (var i = 0; i < codes.length; i++) {
        codes[i].value = "";
      }
      codes[0].focus();
    } else {
      codes[idx].value = "";
      codes[idx].value = e.key;
      e.preventDefault();
    }
  });
});
```

**Comment by kunal00000** [comment]
yes, it worked flawlessly.
Thanks , I got to learn new stuffs from this.
maxlength , keyup keydown , preventDefault properties and working. 👍🏻

**Comment by DevilsAutumn** [comment]
> yes, it worked flawlessly. Thanks , I got to learn new stuffs from this. maxlength , keyup keydown , preventDefault properties and working. 👍🏻

Thats great!

**Review by DevilsAutumn — APPROVED** [review]

**Pull request merged**